stonesoup/tracker/gmphd.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+from scipy.stats import multivariate_normal
+
+from .gaussianmixture import GaussianMixtureMultiTargetTracker
+from ..types import TaggedWeightedGaussianState
+
+
+class GMPHDTargetTracker(GaussianMixtureMultiTargetTracker):
+    """A implementation of the Gaussian Mixture
+    Probability Hypothesis Density (GM-PHD) multi-target filter
+
+    References
+    ----------
+
+    .. [1] B.-N. Vo and W.-K. Ma, “The Gaussian Mixture Probability Hypothesis
+    Density Filter,” Signal Processing,IEEE Transactions on, vol. 54, no. 11,
+    pp. 4091–4104, 2006..
+    """
+
+    def predict(self, timestamp, control_input=None, **kwargs):
+        """
+        Predicts the current components in the
+        :state:`GaussianMixtureState` according to the GM-PHD prediction
+        step.
+
+        Parameters
+        ==========
+        self : :state:`GMPHDTargetTracker`
+            Current GMPHD Tracker at time :math:`k`
+
+        Returns
+        =======
+        self : :state:`GMPHDTargetTracker`
+            GMPHD Tracker with predicted components to time :math:`k | k+1`
+
+        Note
+        ======
+        This is an iteration over a list (:class:`GaussianMixtureState`).
+        It predicts each component :math:`i` according to the underlying
+        :class:`Predictor` class and multiplies its weight :math:`w_i`
+        by the probability of survival :math:`P_s`
+        """
+        if len(self.gaussian_mixture) > 0:
+            for i in range(len(self.gaussian_mixture)):
+                gaussian_state_prediction = self.predictor.predict(
+                    prior=self.gaussian_mixture[i],
+                    control_input=control_input,
+                    timestamp=timestamp,
+                )
+                self.gaussian_mixture[i] = TaggedWeightedGaussianState(
+                    tag=self.gaussian_mixture[i].tag,
+                    weight=self.gaussian_mixture[i].weight *
+                    self.prob_of_survival,
+                    state_vector=gaussian_state_prediction.state_vector,
+                    covar=gaussian_state_prediction.covar,
+                    timestamp=timestamp
+                )
+        # Birth component simply gets added to the components
+        self.birth_component.timestamp = timestamp
+        self.gaussian_mixture.append(self.birth_component)
+
+    def update(self, measurements, timestamp, **kwargs):
```

Comment:
So I believe you can put this logic into a `GaussianMixtureUpdater`, and then this will mean the main tracker can have a similar logic to the regular tracker.